test/Passes/pass-combine-xsmm.mlir

```diff
@@ -0,0 +1,147 @@
+//RUN: tpp-opt %s -combine-xsmm-op-optimization -split-input-file | FileCheck %s
+
+  func.func @entry1(%arg0: memref<8x32x32x32xf32>, %arg1: memref<32x32x32x32xf32>, %arg2: memref<1024xf32>, %arg3: memref<8x32x32x32xf32>, %arg4: memref<32x32x32x32xf32>, %arg5: memref<1024xf32>, %arg6: memref<8x32x32x32xf32>, %arg7: memref<32x32x32x32xf32>, %arg8: memref<1024xf32>, %arg9: memref<8x32x32x32xf32>) {
```

Comment:
The tests are confusing. Please add a comment to what they're testing and name them appropriately.

You need to test all the variations I mention above for both matches and non-matches.

Reply:
Looking again, I do not understand these tests at all. 

The `col0` vs. `col1` tests only change the flag, not the arguments. Aren't they supposed to mark which argument (first or second) is the one being broadcasted?

There are multiple variations of tests with more than one flags but only the first is ever checked. `entry4` has one flag but I don't understand what is the difference or why it should fail. The shape is wrong for `row` broadcast, but that's not what the check is doing, it's only checking for `row` bcast.

You need tests for { `col0`, `col1`, `row0`, `row1` } and set the appropriate shape (`<1024xf32>` and `<256x1xf32>`) and match all of those. Having `<1024xf32>` and `bcast_row_in0` with `<256x1024xf3>` should be a verification error (see #779 and #581).

`entry5` is one case where it should not match, repeating the `relu` is another and needs a test.